pallets/nft/src/lib.rs

```diff
@@ -0,0 +1,456 @@
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create NFT class, tokens belong to the class.
+		///
+		/// - `metadata`: external metadata
+		/// - `properties`: class property, include `Transferable` `Burnable`
+		#[pallet::weight(<T as Config>::WeightInfo::create_class())]
+		#[transactional]
+		pub fn create_class(
+			origin: OriginFor<T>, 
+			metadata: CID, 
+			properties: Properties, 
+			start_block: Option<BlockNumberOf<T>>,
+			end_block: Option<BlockNumberOf<T>>,
+			class_type: ClassType<ClassIdOf<T>>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let next_id = orml_nft::Pallet::<T>::next_class_id();
+
+			// TODO charge
+
+			match class_type {
+				ClassType::Merge(id1, id2, burn) => {
+					if !burn {
+						ensure!(<orml_nft::Module<T>>::classes(id1).is_some(), Error::<T>::ClassIdNotFound);
+						ensure!(<orml_nft::Module<T>>::classes(id2).is_some(), Error::<T>::ClassIdNotFound);
+					} else {
+						let class_info1 = orml_nft::Pallet::<T>::classes(id1).ok_or(Error::<T>::ClassIdNotFound)?;
+						let class_info2 = orml_nft::Pallet::<T>::classes(id2).ok_or(Error::<T>::ClassIdNotFound)?;
+	
+						let data1 = class_info1.data;
+						ensure!(
+							data1.properties.0.contains(ClassProperty::Burnable),
+							Error::<T>::NonBurnable
+						);
+						let data2 = class_info2.data;
+						ensure!(
+							data2.properties.0.contains(ClassProperty::Burnable),
+							Error::<T>::NonBurnable
+						);
+					}
+				}
+				_ => {}
+			}
+
+			let data = ClassData {
+				properties,
+				start_block,
+				end_block,
+				class_type,
+			};
+			orml_nft::Pallet::<T>::create_class(&who, metadata, data)?;
+
+			Self::deposit_event(Event::CreatedClass(who, next_id));
+			Ok(().into())
+		}
+
+		/// Mint NFT token
+		///
+		/// - `to`: the token owner's account
+		/// - `class_id`: token belong to the class id
+		/// - `metadata`: external metadata
+		/// - `quantity`: token quantity
+		#[pallet::weight(<T as Config>::WeightInfo::mint(*quantity))]
+		#[transactional]
+		pub fn mint(
+			origin: OriginFor<T>,
+			to: <T::Lookup as StaticLookup>::Source,
+			class_id: ClassIdOf<T>,
+			metadata: CID,
+			quantity: u32,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let to = T::Lookup::lookup(to)?;
+			ensure!(quantity >= 1, Error::<T>::InvalidQuantity);
+			let class_info = orml_nft::Pallet::<T>::classes(class_id).ok_or(Error::<T>::ClassIdNotFound)?;
+			ensure!(who == class_info.owner, Error::<T>::NoPermission);
+			ensure!(Self::check_time(&class_info.data), Error::<T>::OutOfCampaignPeriod);
+
+			match class_info.data.class_type {
+				ClassType::Simple(max_num) => {
+					let issued = class_info.total_issuance;
+					if TokenIdOf::<T>::from(quantity) + issued > TokenIdOf::<T>::from(max_num) {
+						Err(Error::<T>::QuantitiyOverflow)?
+					}
+				}
+				_ => {
+					Err(Error::<T>::WrongClassType)?
+				}
+			}
+
+			let data = TokenData {
```

Comment:
Maybe you forget to add TODO here like below what you did. I added it in order not to be forgotten.
```suggestion
                         // TODO: adjustible rarity
			let data = TokenData {
```